lib/puppet/provider/netscaler.rb

```diff
@@ -102,17 +102,26 @@ def netscaler_api_type
   end
 
   # I don't want to use `def state=` because that will be called before flush
-  def set_state(value)
+  def set_state(value, name_key, name_val)
+    message_hash = { name_key => name_val }
+    message_hash = { netscaler_api_type => message_hash }
+    message_hash = message_hash.to_json
+
+    action = nil
+    value = value.upcase
     case value
-    when "ENABLED", "DISABLED"
-      state = value.downcase.chop
-      Puppet::Provider::Netscaler.post("/config/#{netscaler_api_type}/#{resource[:name]}?action=#{state}", {
-        netscaler_api_type => {:name => resource[:name],}
-      }.to_json)
-    when nil
-      # Do nothing
-    else
-      err "Incorrect state: #{value}"
+      when 'ENABLED'
+        action = "enable"
+      when 'DISABLED'
+        action = "disable"
+      when nil
+        # Do nothing
+      else
+        err "Incorrect state: #{value}"
+    end
+
+    if action
+      Puppet::Provider::Netscaler.post("/config/#{netscaler_api_type}", message_hash, {"action" => action})
```

Comment:
Removing the #{netscaler_api_type} from the base URL causes the REST call to fail.
Check out the flush method (for updating) and you'll see the same thing there.